hooks/submit_for_review.py

```diff
@@ -0,0 +1,322 @@
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        path = item.properties["path"]
+
+        # log the accepted file and display a button to reveal it in the fs
+        self.logger.info(
+            "Submit for review plugin accepted: %s" % (path,),
+            extra={
+                "action_show_folder": {
+                    "path": path
+                }
+            }
+        )
+
+        # return the accepted info
+        return {"accepted": True}
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        sg_publish = item.properties["sg_publish_data"]
+        sg_task = self.parent.context.task
+        comment = item.description
+        thumbnail_path = item.get_thumbnail_as_path()
+        progress_cb = lambda *args, **kwargs: None
+        review_submission_app = self.parent.engine.apps.get("tk-multi-reviewsubmission")
+
+        render_path = item.properties.get("path")
+        render_template = item.properties.get("work_template")
+        publish_template = item.properties.get("publish_template")
+        render_path_fields = render_template.get_fields(render_path)
+        first_frame = item.properties.get("first_frame")
+        last_frame = item.properties.get("last_frame")
```

Comment:
This works based on the assumption that first and last frame have been set on the item by the collector right?

If that's the case, we should update the base collector to do this for all sequence types (not just tk-nuke collector). My preference would be to use what is already included on the item by the base collector if these values don't exist. 

You can see we store all the paths on the item here: https://github.com/shotgunsoftware/tk-multi-publish2/blob/45377_context_change/hooks/collector.py#L265. Given those paths, you could extract the frame number via the publish template and then figure out the min/max for the first/last frame. 

That seems like a reasonable fallback if first/last frame aren't set by the Nuke collector. 

Maybe we should table that for now and just create a ticket to add that capability in the future. For now we can simply have `accept()` return `False` if the required properties aren't set on the item. That would include `first_frame`, `last_frame`, and `color_space`. 

Can you verify that those are set at collection time. Now that this plugin is in the app, people can use it without nuke, so be sure to document in here what is needed on the item in order for the item to be accepted by this plugin.